File: shoring/flywheel/src/test/java/com/panopset/tests/flywheel/BracketTest.java
package com.panopset.tests.flywheel;

import static com.panopset.compat.Stringop.BAR;
import static com.panopset.compat.Stringop.FOO;
import static org.junit.jupiter.api.Assertions.assertEquals;
import java.io.IOException;
import org.junit.jupiter.api.Test;
import com.panopset.compat.Logop;
import com.panopset.flywheel.FlywheelBuilder;
import com.panopset.flywheel.LineFeedRules;

/**
 * Tests the Flywheel template variable resolution.
 */
public final class BracketTest {

  /**
   * <b>$&#123;1} ${foo}</b>.
   */
  static final String INPUT = "$&#123;1} ${" + FOO + "}.";

  /**
   * <b>$&#123;1} bar</b>.
   */
  static final String EXPECTED = "$&#123;1} " + BAR + ".";

  @Test
  void testBracket() throws IOException {
    Logop.turnOnDebugging();
    final String s = new FlywheelBuilder().withLineFeedRules(LineFeedRules.FLATTEN).map(FOO, BAR)
        .input(new String[] {EXPECTED}).construct().exec();
    assertEquals(EXPECTED, s);
    Logop.turnOffDebugging();
  }

}

File: code/shoring/flywheel/src/test/kotlin/com/panopset/tests/flywheel/BracketTest.kt
package com.panopset.tests.flywheel

import com.panopset.compat.Stringop.BAR
import com.panopset.compat.Stringop.FOO
import com.panopset.flywheel.FlywheelBuilder
import com.panopset.flywheel.LFR_FLATTEN
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

/**
 * Tests the Flywheel template variable resolution.
 */
class BracketTest {
    @Test
    fun testBracket() {
        val s = FlywheelBuilder().withLineFeedRules(LFR_FLATTEN).map(FOO, BAR)
            .inputArray(arrayOf(INPUT)).construct().exec()
        Assertions.assertEquals(EXPECTED, s)
    }
}

val INPUT = "$&#123;1} \${$FOO}."

/**
 * **$&#123;1} bar**.
 */
val EXPECTED = "$&#123;1} $BAR."

